cime_config/buildnml

```diff
@@ -179,6 +179,64 @@ def buildnml(case, caseroot, compname):
             logger.debug("WW3 namelist copy: file1 %s file2 %s ", file1, file2)
             shutil.copy2(file1, file2)
 
+    _prestage_inputs(case)
+            
+###############################################################################
+def _prestage_inputs(case):
+###############################################################################
+    #----------------------------------------------------
+    # Prestage necessary files to rundir
+    #----------------------------------------------------
+    rundir = case.get_value("RUNDIR")
+
+    # Create rundir/ww3_moddef_create
+    filename = "ww3_grid"
+    bldroot = os.path.join(case.get_value("EXEROOT"),"wav","obj")
+    item = os.path.join(bldroot, filename)
+    if os.path.isfile(item):
+        ww3_moddef_dir = os.path.join(rundir, "ww3_moddef_create")
+        if not os.path.exists(ww3_moddef_dir):
+            os.makedirs(ww3_moddef_dir)
+        if not os.path.exists(os.path.join(ww3_moddef_dir,filename)):
+            safe_copy(item, ww3_moddef_dir)
+
+    if case.get_value("WW3_MODDEF") == 'unset':
+        # Create output dir ww3_moddef_create if appropriate
+        output_dir = os.path.join(rundir, "ww3_moddef_create")
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+
+        # Copy ww3_inp and other needed info to ww3_moddef_create directory
+        input_dir = case.get_value("WW3_GRID_INP_DIR")
+        files = os.listdir(input_dir)
+        for filename in files:
+            if not os.path.isfile(os.path.join(output_dir, filename)):
+                safe_copy(os.path.join(input_dir, filename), os.path.join(output_dir, filename))
+                
+        # Create mod_def file using ww3_grid and the grid_input files
+        run_cmd("ww3_grid", from_dir=output_dir)
```

Comment:
Modifying this line as follows should fix this PR:

```run_cmd("./ww3_grid > mod_def.ww3.log", from_dir=output_dir)```

Reply:
Better yet, I suggest replacing lines 218-223 with the following:

```
        if os.path.isfile(os.path.join(output_dir,"ww3_grid")): 
            run_cmd("./ww3_grid > mod_def.ww3.log", from_dir=output_dir)
            if not os.path.isfile(os.path.join(output_dir,"mod_def.ww3")):
                raise RuntimeError("mod_def.ww3 was not created, check mod_def.ww3.log for errors.")
            shutil.move(os.path.join(output_dir,"mod_def.ww3"), os.path.join(rundir, "mod_def.ww3"))
        else:
            logger.warning("ww3_grid file not found. The mod_def.ww3 file will be created after the build phase.")
```

Reply:
Actually, even these changes don't fix the issue with TEST runs. It appears like, when you run a TEST case, the buildnml script doesn't get called in between build and run phases. And so the mod_def file never gets created in a TEST run.

Reply:
I have found the toggle to skip the pnl in tests and am testing a change to unset this when ww3 is in the compset.